mlflow/protos/databricks.proto

```diff
@@ -1,3 +1,5 @@
+syntax = "proto2";
```

Comment:
Q: why this change?

Reply:
Hi :) 
Otherwise we get this warning....only for that reason
[libprotobuf WARNING google/protobuf/compiler/parser.cc:562] No syntax specified for the proto file: databricks.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)